images/testing.go

```diff
@@ -23,6 +28,46 @@ func (f *FakeProvider) GetImage(_ context.Context, token string) (*Image, error)
 	return nil, ErrImageNotFound
 }
 
+// GetImageURL returns the URL of the image associated with a given alert.
+func (f *FakeProvider) GetImageURL(_ context.Context, alert types.Alert) (string, error) {
+	uri, err := getImageURI(alert)
+	if err != nil {
+		return "", err
+	}
+
+	for _, img := range f.Images {
+		if img.Token == uri || img.URL == uri {
+			return img.URL, nil
+		}
+	}
+	return "", ErrImageNotFound
+}
+
+// GetRawImage returns an io.Reader to read the bytes of the image associated with a given alert.
```

Comment:
As part of the comment description, if you're returning a named error is good to let the user of this function know they can assert on the specific error. 

So my suggestion is:

```
// GetRawImage returns an io.Reader to read the bytes of the image associated with a given alert. Returns `ErrImageNotFound` if no image is found.
```